client/components/withModal.jsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import Modal from './Modal.jsx';
+
+const withModal = WrappedComponent => {
```

Comment:
Why are you passing in a component here instead of passing something into this component through props? I'm not saying any of this is wrong, I've just never seen that style before. Could you walk me through this when you have time?